src/main/java/org/mockito/internal/verification/MockAwareVerificationMode.java

```diff
@@ -4,21 +4,44 @@
  */
 package org.mockito.internal.verification;
 
+import java.util.Set;
+
 import org.mockito.internal.verification.api.VerificationData;
+
+import org.mockito.listeners.VerificationListener;
+import org.mockito.verification.VerificationEvent;
 import org.mockito.verification.VerificationMode;
 
 public class MockAwareVerificationMode implements VerificationMode {
 
     private final Object mock;
     private final VerificationMode mode;
+    private final Set<VerificationListener> listeners;
 
-    public MockAwareVerificationMode(Object mock, VerificationMode mode) {
+    public MockAwareVerificationMode(Object mock, VerificationMode mode, Set<VerificationListener> listeners) {
         this.mock = mock;
         this.mode = mode;
+        this.listeners = listeners;
     }
 
     public void verify(VerificationData data) {
-        mode.verify(data);
+        try {
+            mode.verify(data);
+            notifyExceptionListeners(new VerificationEventImpl(mock, mode, data, null));
```

Comment:
Sad that we can't use `Optional` for this. Let's leave a TODO for Mockito 3 to refactor this to `Optional`